### PR TITLE
test: add e2e test cases for route with plugin cors

### LINKED ISSUE #863: bug: 2.1 version create route with cors will cause an error
# Bug report

## Describe the bug
2.1 version create route with `cors` plugin and `cors` used the default setting `{}`, as below：
```
{"name":"test","uris":["/name"],"methods":["GET","HEAD","POST","PUT","DELETE","OPTIONS","PATCH"],"plugins":{"cors":{}},"vars":[],"upstream":{"type":"roundrobin","timeout":{"connect":6000,"send":6000,"read":6000},"nodes":[{"host":"127.0.0.1","port":7000,"weight":1}]}}
```
when request the route, will cause an error:
```
*104794 failed to run header_filter_by_lua*: /Users/idbeta/1106/apisix/apisix/plugins/cors.lua:89: bad argument #1 to 'str_find' (string expected, got nil)
stack traceback:
	[C]: in function 'str_find'
	/Users/idbeta/1106/apisix/apisix/plugins/cors.lua:89: in function 'create_obj_fun'
	/Users/idbeta/1106/apisix/apisix/core/lrucache.lua:92: in function 'plugin_ctx'
	/Users/idbeta/1106/apisix/apisix/plugins/cors.lua:168: in function 'phase_func'
	/Users/idbeta/1106/apisix/apisix/init.lua:159: in function 'run_plugin'
	/Users/idbeta/1106/apisix/apisix/init.lua:622: in function 'common_phase'
	/Users/idbeta/1106/apisix/apisix/init.lua:632: in function 'http_header_filter_phase'
	header_filter_by_lua:2: in main chunk while reading response header from upstream, client: 127.0.0.1, server: , request: "GET /name HTTP/1.1", upstream: "http://127.0.0.1:7000/name", host: "127.0.0.1:9080"
```

### PR DESCRIPTION
**Pull request by nic-chen**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [x] E2E test

- Related issues

close #863

bug has been fixed in APISIX, just test for it.

### CODE REVIEW

File: api/test/e2e/route_with_plugin_cors_test.go [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+		},
+		{
+			caseDesc: "update route with specified setting",
+			Object:   MangerApiExpect(t),
+			Method:   http.MethodPut,
+			Path:     "/apisix/admin/routes/r1",
+			Body: `{
+				"uri": "/hello",
+				"plugins": {
+                    "cors": {
```
Comment (membphis) [inline api/test/e2e/route_with_plugin_cors_test.go:77]: bad indentation

File: api/test/e2e/route_with_plugin_cors_test.go [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+		},
+		{
+			caseDesc: "update route with cors setting force wildcard",
+			Object:   MangerApiExpect(t),
+			Method:   http.MethodPut,
+			Path:     "/apisix/admin/routes/r1",
+			Body: `{
+				"uri": "/hello",
+				"plugins": {
+                        "cors": {
```
Comment (membphis) [inline api/test/e2e/route_with_plugin_cors_test.go:145]: ditto

**Review by membphis — APPROVED** [review]
LGTM, except minor indentation things

File: api/test/e2e/route_with_plugin_cors_test.go (hunk 1 above)
Comment (nic-chen) [reply api/test/e2e/route_with_plugin_cors_test.go]: fixed.

File: api/test/e2e/route_with_plugin_cors_test.go (hunk 2 above)
Comment (nic-chen) [reply api/test/e2e/route_with_plugin_cors_test.go]: fixed.

File: api/test/e2e/route_with_plugin_cors_test.go [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+				"Access-Control-Expose-Headers": "ex-headr1,ex-headr2",
+				"Access-Control-Max-Age":        "50",
+			},
+			ExpectBody: "hello world",
+			Sleep:      sleepTime,
+		},
+		{
+			caseDesc: "verify route with cors specified no match origin",
+			Object:   APISIXExpect(t),
+			Method:   http.MethodGet,
```
Comment (gxthrj) [inline api/test/e2e/route_with_plugin_cors_test.go:121]: I think we need to do some tests on option detection scenarios ，which is useful under non-simple requests.

Comment (nic-chen) [reply api/test/e2e/route_with_plugin_cors_test.go:121]: updated. @gxthrj

Comment (gxthrj) [reply api/test/e2e/route_with_plugin_cors_test.go:121]: Sorry, I mean we had better to test the options method， it should return the `allow*` headers .

Comment (nic-chen) [reply api/test/e2e/route_with_plugin_cors_test.go:121]: @gxthrj 

it return empty: 
https://github.com/apache/apisix/blob/master/t/plugin/cors.t#L427

File: api/test/e2e/route_with_plugin_cors_test.go [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+			Path:     "/hello",
+			Headers: map[string]string{
+				"Origin": "http://sub3.domain.com",
+			},
+			ExpectStatus: http.StatusOK,
+			ExpectHeaders: map[string]string{
+				"Access-Control-Allow-Origin":   "",
```
Comment (gxthrj) [inline api/test/e2e/route_with_plugin_cors_test.go:128]: It is wrong. If we return all headers with empty or no headers, the browser will block the request.

Ref to :https://developer.mozilla.org/zh-CN/docs/Web/HTTP/Access_control_CORS.

Comment (nic-chen) [reply api/test/e2e/route_with_plugin_cors_test.go:128]: updated.

**Review by gxthrj — APPROVED** [review]

**Pull request merged**